src/media-segment-request.js

```diff
@@ -289,7 +289,7 @@ const transmuxAndNotify = ({
 
     if (probeResult) {
       trackInfoFn(segment, {
-        hasAudio: probeResult.hasAudio,
+        hasAudio: !isMuxed && probeResult.hasAudio,
```

Comment:
It seems a bit strange to me that we are reporting that we don't have audio even though we do. Would it be better to perform the check for isMuxed and hasAudio and separating the behavior at a higher level, so we always report what we do see in the result?

Reply:
It is a bit strange, I agree, but I don't know if we want to do that or not. I am open to a refactor but lets go over the reasons I have to keep it right now. 

The reason we have it setup like this is that we will only ever append to and create a `videoBuffer` in these situations as `videoBuffer` will have both codecs: `video/mp4;codecs="avc1.4d400d,mp4a.40.2"`. 
This means that our `videoBuffer` is more of a combined buffer. The segment loader, and anyone else shouldn't care that the segment has audio, since its just a big bundle of data that will only ever be treated as video.  We won't ever see any audio data and the `audioBuffer` won't ever be created.  All data will be spit out by the muxer will [reported as `combined` and changed to `video` ](https://github.com/videojs/http-streaming/blob/e0e497f7ca00aa061147e4dd005d03cb802f9cec/src/media-segment-request.js#L317)

If we want to make such a change I have the code done. I'm just not sure what is best here.

https://github.com/videojs/http-streaming/compare/fix/muxed-no-has-audio...wip/is-muxed?expand=1

Reply:
We should add in @gkatsev too, but I'm inclined to say that if we make changes like this, we should probably do it now. It's hard to read the code as `hasAudio` having the meaning that it's audio only, rather than a component of muxed.

Reply:
are you saying to rename `hasAudio` to something like `audioOnlyMedia`?

Reply:
Probably more of what Brandon prepared in the link, that hasAudio and hasVideo remain descriptors of if the media has audio and video, and isMuxed is passed along to describe whether the content is muxed together or not.

Reply:
Sounds good to me.

Reply:
done